pkg/auth/api/api.go

```diff
@@ -390,3 +401,113 @@ func (s *authService) hasSystemAdminOrganization(orgs []*envproto.Organization)
+					Id:          s.config.DemoSignInConfig.Organization,
+					Name:        "Demo organization",
+					UrlCode:     "demo",
+					Description: "This organization is for demo users",
+					Disabled:    false,
+					Archived:    false,
+					Trial:       false,
+					CreatedAt:   now.Unix(),
+					UpdatedAt:   now.Unix(),
+					SystemAdmin: false,
+				}})
+			if err != nil {
+				return err
+			}
+		}
+		projectStorage := envstotage.NewProjectStorage(tx)
+		_, err = projectStorage.GetProject(ctx, s.config.DemoSignInConfig.Project)
+		if err != nil && errors.Is(err, envstotage.ErrProjectNotFound) {
+			err = projectStorage.CreateProject(ctx, &envdomain.Project{
+				Project: &envproto.Project{
+					Id:             s.config.DemoSignInConfig.Project,
+					Description:    "This project is for demo users",
+					Disabled:       false,
+					Trial:          false,
+					CreatorEmail:   s.config.DemoSignInConfig.Email,
+					CreatedAt:      now.Unix(),
+					UpdatedAt:      now.Unix(),
+					Name:           "Demo",
+					UrlCode:        "demo",
+					OrganizationId: s.config.DemoSignInConfig.Organization,
+				}})
+			if err != nil {
+				return err
+			}
+		}
+		environmentStorage := envstotage.NewEnvironmentStorage(tx)
+		_, err = environmentStorage.GetEnvironmentV2(ctx, s.config.DemoSignInConfig.Environment)
+		if err != nil && errors.Is(err, envstotage.ErrEnvironmentNotFound) {
+			err = environmentStorage.CreateEnvironmentV2(ctx, &envdomain.EnvironmentV2{
+				EnvironmentV2: &envproto.EnvironmentV2{
+					Id:             s.config.DemoSignInConfig.Environment,
+					Name:           "Demo",
+					UrlCode:        "demo",
+					Description:    "This environment is for demo users",
+					ProjectId:      s.config.DemoSignInConfig.Project,
+					Archived:       false,
+					CreatedAt:      now.Unix(),
+					UpdatedAt:      now.Unix(),
+					OrganizationId: s.config.DemoSignInConfig.Organization,
+					RequireComment: false,
+				}})
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		s.logger.Error("prepare demo user organization, project and environment error", zap.Error(err))
+		return
+	}
+
+	accountStorage := accountstotage.NewAccountStorage(s.mysqlClient)
+	_, err = accountStorage.GetAccountV2(
+		ctx,
+		s.config.DemoSignInConfig.Email,
+		s.config.DemoSignInConfig.Organization,
+	)
+	if err != nil && errors.Is(err, accountstotage.ErrAccountNotFound) {
+		err = accountStorage.CreateAccountV2(ctx, &domain.AccountV2{
+			AccountV2: &acproto.AccountV2{
+				OrganizationId:   s.config.DemoSignInConfig.Organization,
+				Email:            s.config.DemoSignInConfig.Email,
+				Name:             "demo",
+				OrganizationRole: acproto.AccountV2_Role_Organization_ADMIN,
+				EnvironmentRoles: []*acproto.AccountV2_EnvironmentRole{
+					{
+						EnvironmentId: s.config.DemoSignInConfig.Environment,
+						Role:          acproto.AccountV2_Role_Environment_EDITOR,
+					},
+				},
+			},
+		})
```

Comment:
We decided to use SQL to create demo users because the account gRPC service might not be ready when we start the backend service.